packages/core/contracts/UnirepSocial.sol

```diff
@@ -101,6 +130,167 @@ contract UnirepSocial {
+     * @param epoch The epoch the subsidy belongs to
+     * @param epochKey The epoch key that receives the subsidy
+     * @param amount The amount requesting to be spent
+     */
+    function trySpendSubsidy(
+      uint256 epoch,
+      uint256 subsidyKey,
+      uint256 amount
+    ) private {
+        uint256 spentSubsidy = subsidies[epoch][subsidyKey];
+        assert(spentSubsidy <= subsidy);
+        uint256 remainingSubsidy = subsidy - spentSubsidy;
+        require(amount <= remainingSubsidy, 'Unirep Social: requesting too much subsidy');
+        require(epoch == unirep.currentEpoch(), 'Unirep Social: wrong epoch');
+        subsidies[epoch][subsidyKey] += amount;
+    }
+
+    function verifyNegativeRepProof(uint256[7] memory publicSignals, uint256[8] memory proof) internal view returns (bool) {
+        return negativeReputationVerifier.verifyProof(
+            [proof[0], proof[1]],
+            [[proof[2], proof[3]], [proof[4], proof[5]]],
+            [proof[6], proof[7]],
+            publicSignals
+        );
+    }
+
+    function verifySubsidyKeyProof(uint256[7] memory publicSignals, uint256[8] memory proof) internal view returns (bool) {
+        return subsidyKeyVerifier.verifyProof(
+            [proof[0], proof[1]],
+            [[proof[2], proof[3]], [proof[4], proof[5]]],
+            [proof[6], proof[7]],
+            publicSignals
+        );
+    }
+
+    /**
+     * Accepts a negative reputation proof
+     * publicSignals[0] - GST root
+     * publicSignals[1] - subsidy key
+     * publicSignals[2] - epoch key
+     * publicSignals[3] - epoch
+     * publicSignals[4] - attester id
+     * publicSignals[5] - maxRep
+     **/
+    function getSubsidyAirdrop(
+      uint256[7] memory publicSignals,
+      uint256[8] memory proof
+    ) public {
+        (,,,uint numEpochKeyNoncePerEpoch,uint maxReputationBudget,,,uint attestingFee,,) = unirep.config();
+        require(publicSignals[4] == attesterId, "Unirep Social: submit a proof with different attester ID from Unirep Social");
+
+        // verify the proof
+        require(verifyNegativeRepProof(publicSignals, proof));
+
+        // update the stored subsidy balances
+        uint requestedSubsidy = publicSignals[5]; // the amount proved
+        uint receivedSubsidy = subsidy < requestedSubsidy ? subsidy : requestedSubsidy;
+        uint epoch = publicSignals[3];
+        trySpendSubsidy(epoch, publicSignals[2], receivedSubsidy);
+        require(unirep.globalStateTreeRoots(epoch, publicSignals[0]), "Unirep Social: GST root does not exist in epoch");
+
+        // Submit attestation to receiver's first epoch key
+        Unirep.Attestation memory attestation;
+        attestation.attesterId = attesterId;
+        attestation.posRep = receivedSubsidy;
+        attestation.negRep = 0;
+        unirep.submitAttestation{value: attestingFee}(
+            attestation,
+            publicSignals[2] // first epoch key
+        );
+    }
```

Comment:
should prevent users from requesting subsidy airdrop more than twice.
Because user can always generate negative rep proof (e.g. `-3`) in the same epoch
so if user request for 10 times with `-3` proof, the user can get 30 reputation after user state transition

Reply:
7d6ff15c9348195821478246342dc9af8ca3cd43

Reply:
7d6ff15c9348195821478246342dc9af8ca3cd43